### ISSUE TITLE
Debug hover does not work in some cases

### ISSUE THREAD

**Issue by waderyan** [Microsoft/vscode#24277]
Test Tour MERN https://github.com/Microsoft/vscode/issues/22277

- VSCode Version: Insiders
- OS Version: macOS

Steps to Reproduce:

1. Clone MERN starter kit
2. Start debugging
3. Set a breakpoint in `post.controller.js` on line 13
4. Refresh the app

I expect to be able to hover over variables and inspect. Instead when I hover I don't see anything. 

![image](https://cloud.githubusercontent.com/assets/2449568/24821966/9fc4e6e6-1ba6-11e7-9970-3de0f076262e.png)

**Comment by Spown** [comment]
I think you are trying to [access a variable from another scope](/Microsoft/vscode-node-debug2/issues/79).

**Comment by isidorn** [comment]
Dup of https://github.com/Microsoft/vscode/issues/24520

**Comment by isidorn** [comment]
Reopening because there is some funky behavior shown to me by @weinand that I would like to investigate

**Comment by dotnetengineers** [comment]
The problem seems to resolve if all blank lines and comments are removed from the TS file.

**Comment by isidorn** [comment]
@dotnetengineers that seems highly unlikely

**Comment by dotnetengineers** [comment]
The above mentioned is reproducible with sample code that at https://github.com/dotnetengineers/Issue_-24277.  On app.ts set a breakpoint on Line 39, start debugger, the last variable that displays hover values is schoolBooks on line 27, any variables after that have no hover value.  Remove all blank lines from app.ts and re-run, all variables now have a hover value.

Windows 10.0.14393
VSCode 1.12.0-insider
Node 7.8.0
TSC 2.2.2

**Comment by dotnetengineers** [comment]
Hello Isidor – I’ve posted some sample code that replicates issue #24277 and resolves when empty lines are removed.

**Comment by isidorn** [comment]
@dotnetengineers thanks, I will investigate more next week

**Comment by isidorn** [comment]
@dotnetengineers thanks for aweseme repro steps!

The issue here seems to be that we are getting bad scope ranges from the node2 debug adapter
fyi @roblourens 
This seems to happen because we are hovering over a typescript file and the scope ranges have line numbers with respect to the javascript file.

To work around this issue I made our scope search strategy a bit more flexibile: if none of the scopes satisfies the range we are searching in -> still try to search across all scopes.
This fallback mechanism fixes this issue.

**Comment by OfekA** [comment]
Is this fixed? @isidorn @chrmarti

I opened a new issue, I hope it is ok
#30286

**Comment by weinand** [comment]
There is a workaround in place (as @isidorn mentioned above).
But the underlying issues is probably not fixed.
@roblourens does the inspector protocol node debugger map the scope ranges via source maps?

**Comment by roblourens** [comment]
Yes, when I tried this, it worked fine for me - https://github.com/Microsoft/vscode-chrome-debug/issues/429

**Comment by mikew** [comment]
I'm still this as only half-working with node.

Here it is working for an object:

<img width="395" alt="vscode node debugger working" src="https://user-images.githubusercontent.com/4729/32966659-edfe15b6-cbb0-11e7-8ac1-9feb426a10d7.png">

But then I don't get much else, here I am hovering over `isWhitelisted`:

<img width="302" alt="vscode node debugger not working 1" src="https://user-images.githubusercontent.com/4729/32966673-fabdd516-cbb0-11e7-862d-fd5f20d25b2c.png">

And again nothing for `req` or `req.path`:

<img width="327" alt="vscode node debugger not working 2" src="https://user-images.githubusercontent.com/4729/32966682-035dba9c-cbb1-11e7-9252-249c416b3410.png">

This is our launch configuration:

```json
    {
      "type": "node",
      "request": "attach",
      "name": "Attach To Running Server",
      "restart": true,
      "sourceMaps": true,
      "showAsyncStacks": true,
      "port": 9231
    },
```

**Comment by roblourens** [comment]
Are you using babel?

**Comment by mikew** [comment]
Yes, in the worst way possible: requiring `babel-register`.

**Comment by roblourens** [comment]
(We just activated the "lock old issues" bot, nothing personal 😄 )

That is kind of unavoidable right now but will be fixed by #37838
